domain/network/service/container_test.go

```diff
@@ -276,10 +278,150 @@ func (s *containerSuite) TestDevicesToBridgeLocalBridgeReqsUnsatisfiable(c *tc.C
 	c.Assert(err, tc.ErrorIs, errors.SpaceRequirementsUnsatisfiable)
 }
 
+func (s *containerSuite) TestDevicesForGuestBridgeFoundNoContainerAddresses(c *tc.C) {
+	defer s.setupMocks(c).Finish()
+
+	s.setupServiceAndMachines(c)
+
+	ctx := c.Context()
+
+	spaceUUID := "positive-space-uuid"
+	spaces := []internal.SpaceName{{
+		UUID: spaceUUID,
+		Name: "positive-space",
+	}}
+	bridgeName := "br-eth0"
+	cidr := "10.10.10.0/24"
+
+	exp := s.st.EXPECT()
+	exp.GetMachineSpaceConstraints(ctx, s.guestUUID.String()).Return(spaces, nil, nil)
+	exp.GetMachineAppBindings(ctx, s.guestUUID.String()).Return(nil, nil)
+	exp.GetMachineNetNodeUUID(ctx, s.hostUUID.String()).Return(s.nodeUUID, nil)
+	// A bridge in the space means that connectivity is satisfied.
+	exp.NICsInSpaces(ctx, s.nodeUUID).Return(map[string][]network.NetInterface{
+		spaceUUID: {
+			{
+				Name: bridgeName,
+				Type: corenetwork.BridgeDevice,
+			},
+		},
+	}, nil)
+	exp.GetSubnetCIDRForDevice(ctx, s.nodeUUID, bridgeName, spaceUUID).Return(cidr, nil)
+
+	s.providerWithNetworking.EXPECT().SupportsContainerAddresses(ctx).Return(false, nil)
+
+	nics, err := s.svc.DevicesForGuest(ctx, s.hostUUID, s.guestUUID)
+	c.Assert(err, tc.ErrorIsNil)
+
+	c.Assert(nics, tc.HasLen, 1)
+	nic := nics[0]
+
+	c.Check(nic.Name, tc.Equals, "eth0")
+	c.Check(nic.MACAddress, tc.NotNil)
+	c.Check(nic.Type, tc.Equals, corenetwork.EthernetDevice)
+	c.Check(nic.ParentDeviceName, tc.Equals, bridgeName)
+	c.Check(nic.IsEnabled, tc.IsTrue)
+	c.Check(nic.IsAutoStart, tc.IsTrue)
```

Comment:
suggestion: `tc.NewMultiChecker()` might be helpful here. Ignore the parts not needed.